src/gt4py/next/program_processors/runners/dace_fieldview/gtir_dataflow.py

```diff
@@ -83,6 +76,83 @@ class IteratorExpr:
     indices: dict[gtx_common.Dimension, ValueExpr]
 
 
+class DataflowInputEdge(Protocol):
+    """Allows to setup an input edge in a map scope."""
+
+    @abc.abstractmethod
+    def connect(self, me: dace.nodes.MapEntry) -> None:
+        pass
+
+
+@dataclasses.dataclass(frozen=True)
+class DataflowMemletEdge(DataflowInputEdge):
+    """
+    Allows to setup an input memlet through a map entry node.
+
+    The edge source has to be a data access node, while the destination node can either
+    be a tasklet, in which case the connector name is also required, or an access node.
+    """
+
+    state: dace.SDFGState
+    source: dace.nodes.AccessNode
+    subset: sbs.Range
+    dest: dace.nodes.AccessNode | dace.nodes.Tasklet
+    dest_conn: Optional[str]
+
+    def connect(self, me: dace.nodes.MapEntry) -> None:
+        memlet = dace.Memlet(data=self.source.data, subset=self.subset)
+        self.state.add_memlet_path(
+            self.source,
+            me,
+            self.dest,
+            dst_conn=self.dest_conn,
+            memlet=memlet,
+        )
+
+
+@dataclasses.dataclass(frozen=True)
+class DataflowEmptyEdge(DataflowInputEdge):
+    """Allows to setup an input edge from a map entry node to a tasklet withough arguements."""
+
+    state: dace.SDFGState
+    node: dace.nodes.Tasklet
+
+    def connect(self, me: dace.nodes.MapEntry) -> None:
+        self.state.add_nedge(me, self.node, dace.Memlet())
+
+
+@dataclasses.dataclass(frozen=True)
+class DataflowOutputEdge:
+    """Allows to setup a data edge to write the result through a map exit node."""
```

Comment:
Also a bit more explanation needed how it works and how it is intended to be used.